src/fu-main.c

```diff
@@ -462,12 +472,18 @@ fu_main_authorize_install_queue (FuMainAuthHelper *helper_ref)
 		return;
 	}
 
+	signal_handler = g_unix_signal_add_full (G_PRIORITY_DEFAULT,
```

Comment:
I did imagine this in main() and then deciding what to do in the signal handler -- what happens if we do SIGTERM and there's not an update happening? Should we just exit straight away? SIGKILL from systemd wouldn't be awesome.

Reply:
The typical thing to do in a `SIGTERM` handler is to do cleanup (such as flush buffers, close descriptors etc).

I would expect if not in an update to exit immediately on `SIGTERM`.
What sort of other cleanup would you expect to happen on `SIGTERM` while not in an update?

And you're right while in an update if `SIGTERM` is received may need to cache the need to close until update is done, so the handler may need to grow.

Reply:
adjusted